### PR TITLE
[PINOT-3301]: Fix broken backward compatibility when reading older star-tree segments.

### PR DESCRIPTION
**Pull request by mayankshriv**
After the StarTreeIndexNode data-structure was modified to add new
members, older segments were no longer being read correctly, even though
these fields did have a default value in the code. Until we design our
own on-disk format for star-tree along with its own
serializer/de-serializer, we can interpret any aggregatedDocId value <
numRawDocs as invalid (indicating older segment).

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
LGTM.

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/StarTreeIndexOperator.java [hunk 1]
```diff
@@ -360,7 +377,8 @@ public BlockId getId() {
       HashSet<String> remainingPredicateColumns = searchEntry.remainingPredicateColumns;
       HashSet<String> remainingGroupByColumns = searchEntry.remainingGroupByColumns;
       // Check if its leaf
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/StarTreeIndexOperator.java]: update the comment, its terminating if its leaf or (all filters are applied and aggregatedDocId is valid)

**Comment by kishoreg** [comment]
LGTM

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/StarTreeIndexOperator.java (hunk 1 above)
Comment (mayankshriv) [inline pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/StarTreeIndexOperator.java]: Updated the comment.

**Pull request merged**